extensions/ql-vscode/docs/test-plan.md

```diff
@@ -0,0 +1,257 @@
+2. Run it against the "Top 10" repositories.
+3. Check the notification message. It should:
+   - Show the number of repos that are going to be queried
+   - Provide a link to the actions workflow
+4. Check the query history. It should:
+   - Show that an item has been added to the query history
+   - The item should be marked as “in progress”.
+5. Once the query completes:
+   - Check that a notification is shown with a link to the results view
+   - Check that the results are rendered with an alert message and a highlighted code snippet:
+![highlighted-code-snippet](images/highlighted-code-snippet.png)
+
+### Test Case 3: Running a non-problem query and viewing results
+
+1. Open the [FunLinesOfCode query](https://github.com/github/codeql/blob/main/cpp/ql/src/Metrics/Functions/FunLinesOfCode.ql).
+2. Run it against a single repository (e.g. `google/brotli`).
+3. Once the query is complete:
+   - Open the query results
+   - Check that the results show up in a table:
+![results-table](images/results-table.png)
+
+### Test Case 4: Interacting with query history
+
+1. Click a history item (for MRVA):
+   - Check that exporting results works
+   - Check that sorting results works
+   - Check that copying repo lists works 
+2. Open the query directory (containing results):
+   - Check that the correct directory is opened and there are results in it
+3. Open variant analysis on GitHub
+   - Check that the correct workflow is opened
+
+### Test Case 5: Cancelling a variant analysis run
+
+When running one of the above queries, try cancelling one from within VS Code:
+- Check that the query is cancelled and the query history item is updated.
+- Check that the workflow run is also cancelled. 
+- Check that any available results are visible in VS Code.
+
+### Test Case 6: Change to a different colour theme 
+
+When running one of the above queries, try changing to a different colour theme 
+and check that everything looks sensible. Are there any components that are not showing up?
+
+## Optional Test Cases
+
+### 1. Selecting repositories to run on
+
+#### Test case 1: Running a query on a single repository
+1. When the repository exists and is public 
+   1. Has a CodeQL database for the correct language 
+   2. Has a CodeQL database for another language 
+   3. Does not have any CodeQL databases
+2. When the repository exists and is private
+   1. Is accessible and has a CodeQL database
+   2. Is not accessible
+3. When the repository does not exist
+
+#### Test case 2: Running a query on a custom repository list
+1. The repository list is non-empty
+   1. All repositories in the list have a CodeQL database 
+   2. Some but not all repositories in the list have a CodeQL database
+   3. No repositories in the list have a CodeQL database
+2. The repository list is empty
+
+#### Test case 3: Running a query on a built-in repository list (i.e. topN)
+- Check that the query runs and results are shown
```

Comment:
This seems to be the same as required test case 2

Reply:
Yeah there's not much difference.